isisd/isis_lsp.c

```diff
@@ -1420,7 +1426,12 @@ int _lsp_regenerate_schedule(struct isis_area *area, int level,
 			"ISIS (%s): Checking whether L%d needs to be scheduled",
 			area->area_tag, lvl);
 
-		if (area->lsp_regenerate_pending[lvl - 1]) {
+		if (area->lsp_regenerate_pending[lvl - 1]
+		    && !(area->bfd_signalled_down)) {
+			/*
+			 * Note: in case of a BFD 'down' message the refresh is
+			 * scheduled once again just to be sure
```

Comment:
Good -- this should bring the session back up if there is link but BFD is just having problems getting packets through ...

Reply:
Yea :)